src/base/memory_monitor_sr.cpp

```diff
@@ -402,29 +490,108 @@ namespace cubperf
     m_mmon = mmon;
   }
 
-  int memory_monitor::aggregater::get_server_info (const MMON_SERVER_INFO &info)
+  void memory_monitor::aggregater::get_server_info (MMON_SERVER_INFO &info) const
   {
-    return 0;
+    strncpy (info.name, m_mmon->m_server_name.c_str (), m_mmon->m_server_name.size ());
+    info.total_mem_usage = m_mmon->m_total_mem_usage.load ();
   }
 
-  int memory_monitor::aggregater::get_module_info (const MMON_MODULE_INFO &info, int module_index)
+  int memory_monitor::aggregater::get_module_info (MMON_MODULE_INFO *&info, int module_index) const
   {
-    return 0;
+    int error = NO_ERROR;
+
+    if (module_index == -1)
+      {
+	// aggregate all detail memory information of modules
+	for (int idx = 0; idx < MMON_MODULE_LAST; idx++)
+	  {
+	    error = m_mmon->m_module[idx]->aggregate_stats (info[idx], false);
+	    if (error != NO_ERROR)
```

Comment:
Don't we have to free allocated memory of succeeded aggregation? For example, 
m_mmon->m_module[0]->aggregate_stats // success
m_mmon->m_module[1]->aggregate_stats // success
m_mmon->m_module[2]->aggregate_stats // error

Or are they gonna addressed in the outside?

Reply:
If the above is the case, depending on the last parameter of `aggregated_stats`, `get_summary`, the caller decides if it should free memory or not. I think it's bad because it can be only warned by a comment. 
To resolve it, it seems that there are two way in handy:
1. Split the function into `aggregate_summary_stats` and `aggregate_stats`.
2. Use a smart pointer so leave it to RAII.

Sorry if I get it wrong.

Reply:
Inside `mmon_module::aggregate_stats()`, there is a deallocation phase for the allocated memory in case of an error. However, this deallocation is for a single module processing, and in the situation where errors occur while aggregating information for multiple modules, deallocation cannot be performed for the previously successful aggregation before the error. To solve this situation, I modify the logic of deallocation that network function, where first allocation of `MMON_XXX_INFO` was occurred, handles all deallocations. With this updated logic, there is no need to perform deallocation for a single module within `mmon_module::aggregate_stats()`, so that part has been removed.